lec06-pop-models.Rmd

```diff
@@ -2,3 +2,360 @@
+
+logistic_eqn <- function(N,r,K) { # calculate dN/dt for the logistic equation
+  return(r*N*(1-N/K))
+}
+
+dN_dt = logistic_eqn(N_seq,r=0.5,K=80)  # notice the vectorized implementation
+
+qplot(N_seq,dN_dt) +
+  geom_line(aes(x=N_seq,y=0*N_seq)) # a line at zero for visual aid
+```
+
+The fixed points are at $N=0$ and $N=80$. If we change $K$, we can figure out that the second fixed point is actually just $N=K$. 
+
+Now we draw the phase portrait (on blackboard):
+
+<!-- Phase portrait for logistic eqn -->
+
+As with the exponential growth model, the fixed point at $0$ is unstable, but now there is also a 
+fixed point at $K$ which is stable. This means that at long times, the system will go towards $N=K$. 
+
+### Note: the importance of stability
+
+Why is stability important? The real world is noisy, and while mathematically an unstable fixed point is 
+still a fixed point (if you were exactly there you would stay there forever), in reality there will 
+always be small perturbations (perhaps a member of the population dies or migrates in) that will move 
+you away from the fixed point. The important thing to know, then, is whether the population will remain 
+close to that fixed point or move away to a different point. 
+
+Think of a ball balanced perfectly on top of a hill, or think of balancing a pen on your finger, 
+or think of balancing a chair on the back two legs. It can stay there as long as there are no disturbances, 
+but the tiniest breath of wind or tremble in your balance will knock it over and it will go towards its other
+fixed point - the ground.
+
+### Note: discrete vs. continuous
+
+There is a subtlety to these differential equation descriptions of models. To see this, calculate
+$\frac{dN}{dt}$ for $r=1$ and $N=0.1$. Is this value positive, negative, or zero? Think about what $N=0.1$
+means biologically. Does it make sense that the population should increase if $N=0.1$?
+
+```{r}
+r <- 1
+N <- 0.1
+
+print(r*N)
+```
+
+Clearly, if $N$ represents a real population, it can only ever take integer values (it must be *discrete*), 
+and it can never be negative. But there's nothing in the model by itself, $\frac{dN}{dt} = rN$, that requires
+any of those things. Models like this are *continuous* - the variable $N$ can be any real number and isn't
+restricted to integers. 
+
+How do we reconcile this with the real world? For "large" population sizes, the continuous description is
+fairly accurate, since if $N$ is, say, 400, then a change of $\pm 1$ is a relatively small change, and we can
+think of $N$ as varying continuously. But for small populations, the continuous description breaks down. 
+If $N$ falls below $1$, the continuous description might tell you that it can bounce back, but in reality 
+$N<1$ means the population is extinct. Even though $N=0$ was an unstable fixed point for the systems we 
+looked at so far, it's what is called and **absorbing** state: once you enter it, you can never leave 
```

Comment:
s/and/an/

Reply:
Not sure what this means?

Reply:
search & replace (on this line)

Reply:
and -> an

I still like that syntax more, although it does not give as many geekpoints